### ISSUE TITLE
Setting custom cache options when using MultiNamespacedCacheBuilder?

### ISSUE THREAD

**Issue by tsaarni** [kubernetes-sigs/controller-runtime#2273]
There are two different forms of builders:

* `cache.BuilderWithOptions()` allows setting custom `cache.Options` like `TransformByObject` 
* `cache.MultiNamespacedCacheBuilder()` allows namespace-limited watches

Is it possible to create `NewCacheFunc` that combines both of these features, e.g. by composing the builders?

I see that in `main` branch `MultiNamespacedCacheBuilder` is deprecated, and namespacing will be achieved via `cache.Options` so the conflict seems to be removed.  Do you have initial release plans for the new API?

**Comment by alvaroaleman** [comment]
No, this isn't supported as of today. There is https://github.com/kubernetes-sigs/controller-runtime/pull/2261 which desribes how and what we want to be configurable in the future. Could you have a look there and check if it would fulfill your use-case, please?

**Comment by tsaarni** [comment]
Thanks @alvaroaleman! I've checked the design doc and it looks good to me. I believe it will work fine for our use case. :+1:
